app/models/mixins/miq_provision_mixin.rb

```diff
@@ -63,18 +63,12 @@ def get_owner
     @owner ||= begin
       email = get_option(:owner_email).try(:downcase)
       return if email.blank?
-      requester = miq_request.requester
-      owner = requester.email.to_s.downcase == email ? requester : User.where("lower(email) = ?", email).first
-      owner.tap { |o| set_owner_group(o) }
+      User.find_by_lower_email(email, miq_request.requester).tap { |o| set_owner_group(o) }
     end
   end
 
   def set_owner_group(user)
-    group_id = get_option(:owner_group)
-    if user && group_id.present?
-      group = user.miq_groups.detect { |g| g.description == group_id }
-      user.current_group = group if group
-    end
+    user.miq_group_description = get_option(:owner_group) if user
```

Comment:
I misunderstood, the nil in question was `user`.  Since this is down to one line and there are no other callers maybe this logic just moves into `tap` block in the `get_owner` method and `set_owner_group` goes away.